Source/SwiftLintFramework/Extensions/NSRegularExpression+SwiftLint.swift

```diff
@@ -13,22 +13,32 @@ public typealias NSRegularExpression = RegularExpression
 public typealias NSTextCheckingResult = TextCheckingResult
 #endif
 
-private var regexCache = [String: NSRegularExpression]()
+private var regexCache = [RegexCacheKey: NSRegularExpression]()
+
+private struct RegexCacheKey: Hashable {
+    let pattern: String
+    let options: NSRegularExpression.Options
+
+    var hashValue: Int {
+        return pattern.hashValue ^ options.rawValue.hashValue
+    }
+
+    static func == (lhs: RegexCacheKey, rhs: RegexCacheKey) -> Bool {
+        return lhs.options == rhs.options && lhs.pattern == rhs.pattern
+    }
+}
 
 extension NSRegularExpression {
-    internal static func cached(pattern: String) throws -> NSRegularExpression {
-        if let result = regexCache[pattern] {
+    internal static func cached(pattern: String,
+                                options: Options? = nil) throws -> NSRegularExpression {
+        let options = options ?? [.anchorsMatchLines, .dotMatchesLineSeparators]
+        let key = RegexCacheKey(pattern: pattern, options: options)
+        if let result = regexCache[key] {
             return result
         }
 
-        let result = try NSRegularExpression(pattern: pattern,
-            options: [.anchorsMatchLines, .dotMatchesLineSeparators])
-        regexCache[pattern] = result
+        let result = try NSRegularExpression(pattern: pattern, options: options)
+        regexCache[key] = result
         return result
     }
-
-    internal static func forcePattern(_ pattern: String) -> NSRegularExpression {
```

Comment:
I've removed this since I don't get why we should have it and the free `regex` function if they do the same thing.